### ISSUE TITLE
[BUG REPORT] Fix Currency Property Naming in ExchangeMarket class

### ISSUE THREAD

**Issue by superhappychris** [jjxtra/ExchangeSharp#267]
Right now the ExchangeMarket class uses the property names MarketCurrency and BaseCurrency to represent what is usually called the "base currency" and "quote currency" respectively.  This is extremely confusing and prone to causing bugs.  Just doing a quick search I found that at least the ExchangeAbucoinsAPI and ExchangeKrakenAPI are assigning the incorrect value to these properties...

I think we should rename MarketCurrency to BaseCurrency, and rename BaseCurrency to QuoteCurrency.  I'm happy to make the fix, but I realize this could be a breaking change for some people using this library, which is why I'm mentioning it here for discussion first.  Thoughts?

**Comment by jjxtra** [comment]
Yea this is not consistant. I think there is another part in the framework that has them reversed too. So we should search for all instance of BaseCurrency and make sure they are correct. I am fine with the change if you want to pull request. Until we get to v. 1.0. I am ok with breaking changes.

**Comment by vslee** [comment]
I agree with renaming for consistency with the market. I was not aware of this issue until reading your post. Perhaps there can be a notice placed on the front page for others to see.

**Comment by superhappychris** [comment]
Ok sounds good. I took a closer look at the code and found 5 classes that have inconsistent property names that need to be changed.  I've listed them below by class.  The left hand side is the current name, the right hand side is what they will be renamed to.  Let me know if these look wrong or if there are any I'm missing.  In the meantime, I'll get started making the changes.

- ExchangeMarket
  - BaseCurrency : _QuoteCurrency_
  - MarketCurrency : _BaseCurrency_
- ExchangeTicker
  - BaseSymbol : _QuoteCurrency_
  - BaseVolume : _QuoteVolume_
  - ConvertedSymbol : _BaseCurrency_
  - ConvertedVolume : _BaseVolume_
- ExchangeTransaction
  - Symbol : _Currency_
- ExchangeDepositDetails
  - Symbol : _Currency_
- MarketCandle
  - BaseVolume : _QuoteVolume_
  - ConvertedVolume : _BaseVolume_

**Comment by bichuga** [comment]
Along similar lines, my recollection is there is some inconsistency between
what we call a currrency "BTC" vs a market "FOO/BTC". Sometimes it's
referred to as a Ticker. Other times a Symbol. It would be really nice to
standardize this if it hasn't been yet.

**Comment by superhappychris** [comment]
@bichuga I agree that the naming for tickers and symbols should be consistent, but I'm not sure where this is happening in the code.  A search for "ticker" returns 396 matches and I don't have time to go through them at the moment.  Can anyone take a look at this?

**Comment by jjxtra** [comment]
@superhappychris The exchangemarket I believe is one too. So basically search for anything using "BaseCurrency", "MarketCurrency" and "QuoteCurrency".

**Comment by superhappychris** [comment]
@jjxtra I listed the ExchangeMarket class above already, is there something else you want me to change about it other than what I mentioned before?

**Comment by jjxtra** [comment]
@superhappychris nope!

**Comment by superhappychris** [comment]
The changes have been submitted.

Right now all symbols are still represented as simple strings but I think it would be better if we gave them their own class to encapsulate some of the tricky logic around them, especially when it comes to figuring out their constituent parts (currencies and separator).  I don't have any time to do that right now, but wanted to put the idea out there in case anyone else has the bandwidth.

**Comment by jjxtra** [comment]
Each exchange handles the symbols and pair orders slightly differently. We could encapsulate the logic in a class, but each Exchange would need an instance of the class to deal with the symbol pairs.

Probably long term ideally we could map each exchange symbols internally to normalized, standard global symbols. The exchanges already have methods to do this, but if we change all the API to use the global/normalized symbol instead of exchange specific symbols, that would make life easier on everyone. Right now you can do it by manually converting global/normalized symbol to exchange specific symbol but this seems like a redundant step.

p.s. sorry about breaking your pull request, I was surprised C# compiler could not handle a property setter with the null coalesce operator, but I should have checked before commit. Sorry about that.

**Comment by superhappychris** [comment]
@jjxtra the majority of exchanges all handle them the same way as far as terminology is concerned. Bittrex is the only one that uses the opposite meaning for the terms BaseCurrency / QuoteCurrency.  All other differences are cosmetic in nature (ordering, separator, capitalization).  

I agree that maintaining an internal mapping would be a good thing.  The current methods for normalizing the symbols fall short if the exchange doesn't use a symbol separator, which is quite a few exchanges.  There just isn't enough metadata in a string to figure out the constituent currencies in that case unfortunately.  So we'll either need to hard code the currencies and symbols in those exchanges or encapsulate the necessary metadata in a class.  Using a class would let us avoid the hard coding scenario, which is always a good thing.

and no worries about the pull request, it happens
